### ISSUE TITLE
Studio with image preview crashes if image isn't added yet

### ISSUE THREAD

**Issue by RobbinHabermehl** [sanity-io/sanity#3004]
**Describe the bug**

When including an image preview in a schema, like this for instance:

```
export const articleSchema = {
  title: "Article",
  name: "article",
  preview: {
    select: {
      media: "cover"
    }
  }
};
```

The Sanity Studio will crash if that image hasn't been set yet:

```
Error: Objects are not valid as a React child (found: object with keys {_type, nl}). If you meant to render a collection of children, use an array instead.
    in ObserveForPreview
    in withPropsStream(ObserveForPreview) (created by WithVisibility)
    in span (created by WithVisibility)
    in WithVisibility (created by PreviewSubscriber)
    in PreviewSubscriber (created by PreviewFields)
    in PreviewFields (created by DocumentHeaderTitle)
    in DocumentHeaderTitle (created by DocumentPanelHeader)
    in span (created by styled.span)
    in styled.span (created by ForwardRef(Text))
    in span (created by ForwardRef(Text))
    in div (created by styled.div)
    in styled.div (created by ForwardRef(Text))
    in ForwardRef(Text) (created by Styled(Component))
    in Styled(Component) (created by ForwardRef(PaneHeader))
```

```
   in StructureError (created by DeskToolRoot)
    in DeskToolRoot (created by RenderTool)
    in ErrorBoundary (created by RenderTool)
    in RenderTool (created by SchemaErrorReporter)
    in RouteScope (created by SchemaErrorReporter)
    in div (created by styled.div)
    in styled.div (created by ForwardRef(Box))
    in ForwardRef(Box) (created by Styled(Component))
    in Styled(Component) (created by ForwardRef(Flex))
    in ForwardRef(Flex) (created by Styled(Component))
    in Styled(Component) (created by SchemaErrorReporter)
    in div (created by styled.div)
    in styled.div (created by ForwardRef(Box))
    in ForwardRef(Box) (created by Styled(Component))
    in Styled(Component) (created by ForwardRef(Flex))
    in ForwardRef(Flex) (created by Styled(Component))
    in Styled(Component) (created by SchemaErrorReporter)
    in div (created by styled.div)
    in styled.div (created by ForwardRef(Box))
    in ForwardRef(Box) (created by Styled(Component))
    in Styled(Component) (created by ForwardRef(Flex))
    in ForwardRef(Flex) (created by Styled(Component))
    in Styled(Component) (created by SchemaErrorReporter)
    in SchemaErrorReporter (created by DefaultLayout)
    in DefaultLayout (created by DefaultLayoutRoot)
    in RouterProvider (created by DefaultLayoutRoot)
    in LoginWrapper (created by NormalizedLoginWrapper)
    in NormalizedLoginWrapper (created by DefaultLayoutRoot)
    in DefaultLayoutRoot (created by AppProvider)
    in div (created by styled.div)
    in styled.div (created by ForwardRef(Box))
    in ForwardRef(Box) (created by Styled(Component))
    in Styled(Component) (created by ForwardRef(Card))
    in Fe (created by ThemeProvider)
    in ThemeProvider (created by ThemeColorProvider)
    in ThemeColorProvider (created by ForwardRef(Card))
    in ForwardRef(Card) (created by Styled(Component))
    in Styled(Component) (created by AppProvider)
    in SnackbarProvider (created by AppProvider)
    in ToastProvider (created by AppProvider)
    in LayerProvider (created by AppProvider)
    in PortalProvider (created by AppProvider)
    in UserColorManagerProvider (created by AppProvider)
    in AppProvider (created by SanityRoot)
    in Fe (created by ThemeProvider)
    in ThemeProvider (created by SanityRoot)
    in ZIndexProvider (created by SanityRoot)
    in SanityRoot
    in AppContainer
```

This only occurs if the document has just been created or after a page refresh of that specific document. 

**To Reproduce**

1. Create a schema that has a "media preview".
2. Create a new document of that type or refresh the editor page of a document that doesn't include the referenced image yet.
3. The Studio will crash and an error view will be shown.

**Expected behavior**

In case the referenced image isn't available yet the Studio shouldn't crash. At least you should have the possibility to render an alternative image in the `prepare()` function.

**Which versions of Sanity are you using?**

```
@sanity/cli             2.23.0 (up to date)
@sanity/base            2.23.0 (up to date)
@sanity/cli             2.23.0 (up to date)
@sanity/client          2.23.0 (up to date)
@sanity/code-input      2.23.0 (up to date)
@sanity/color-input     2.23.0 (up to date)
@sanity/components      2.14.0 (up to date)
@sanity/core            2.23.0 (up to date)
@sanity/default-layout  2.23.0 (up to date)
@sanity/default-login   2.23.0 (up to date)
@sanity/desk-tool       2.23.0 (up to date)
```

**What operating system are you using?**

Chrome v96 on MacOS.

**Which versions of Node.js / npm are you running?**

```
6.14.15
v14.18.1
```

**Additional context**

Originally I had v2.10.2 of Sanity configured. Upgrading to the latest versions didn't resolve the issue though.

**Comment by KevinBodi** [comment]
I got this exact same issue, I got it just after updating to Sanity 2.23.1 from 2.11.2 and running on Node v14.18.2.

Do you also have a title-field that is an object of strings? For me, the error stops occurring if I change the title field to be a single string instead of an object of multiple strings.

Both these approaches lead to the crash
```
preview: {
    select: {
      titleNo: 'title.no',
      titleEn: 'title.en'
    },
    prepare({ titleNo, titleEn }) {
      return {
        title: titleNo || titleEn || 'No title'
    }
  }
}
```
```
preview: {
    select: {
      title: 'title'
    },
    prepare({ title }) {
      return {
        title: title.no || title.en || 'No title'
    }
  }
}
```
But more interestingly, the second approach gives a more detailed error in the console before it crashes
```
prepareForPreview.js:79 Heads up! Got error while preparing data for preview. Click for details.

prepareForPreview.js:83 Check the preview config for schema type "article": 
{
    "select": {
        "title": "title",
        "image": "mainImage",
        "date": "_updatedAt",
        "description": "content"
    }
}
prepareForPreview.js:97 Encountered an error when calling prepare(
{
    "title": undefined,
}
```
Both approaches worked without errors before updating.

My error message once it crashes, looks like this:

```
Error: Objects are not valid as a React child (found: object with keys {_type, no}). If you meant to render a collection of children, use an array instead.
    in ObserveForPreview
    in withPropsStream(ObserveForPreview) (created by WithVisibility)
    in span (created by WithVisibility)
    in WithVisibility (created by PreviewSubscriber)
    in PreviewSubscriber (created by PreviewFields)
    in PreviewFields (created by DocumentHeaderTitle)
    in DocumentHeaderTitle (created by ForwardRef)
    in span (created by styled.span)
    in styled.span (created by ForwardRef(Text))
    in span (created by ForwardRef(Text))
    in div (created by styled.div)
    in styled.div (created by ForwardRef(Text))
    in ForwardRef(Text) (created by Styled(Component))
    in Styled(Component) (created by ForwardRef(PaneHeader))
 ```

**Comment by KevinBodi** [comment]
There seems to be some changes done to fields named "title" from your version to mine 2.23.0 -> 2.23.1. https://www.sanity.io/changelog#change-46bed66b-aa6e-42d6-b67d-f02bab21b320

But that fix didn't solve this error. I suspect that the 2.23.x-update introduced a rule somewhere out of our control where `_ref.title` is expected to be of type string/array and can't be an object. One fix would be to upgrade to 2.22.x until Sanity comes with a fix where _ref.title is allowed to be an object, because [this example](https://www.sanity.io/docs/localization#:~:text=Title%27%2C%0A%20%20%20%20%20%20name%3A%20%27title%27%2C-,type%3A%20%27localeString%27,-%7D%0A%20%20%5D%2C%0A%20%20preview%3A%20%7B%0A%20%20%20%20select%3A%20%7B%0A%20%20%20%20%20%20title) shown by Sanity is no longer valid without causing crashes.

**Comment by KevinBodi** [comment]
After extensive testing, I've come to see that all these three conditions cause a crash if `_ref.title` is an object.
```
preview: {
    select: {
      title: 'title'
    }
  }
```
```
preview: {
    select: {
      title: 'title.no',
      media: 'mainImage',
    }
  }
```
```
preview: {
    select: {
      title: 'title.no',
      image: 'mainImage'
    },
    prepare({ title }) {
      return {
        title
      };
    }
  }
```

The only two ways to avoid the crash is again to not have `_ref.title` as an object but string instead or to do this (without media as you described)
```
preview: {
    select: {
      title: 'title.no'
    }
  }
```
which I suppose avoids triggering the prepare()-function but still allow translations.

Sadly these alternatives won't work for my case.

**Comment by bjoerge** [comment]
We're rolling out a fix for this asap. Thanks everyone!

**Comment by bjoerge** [comment]
Fixed in [v2.23.2](https://github.com/sanity-io/sanity/releases/tag/v2.23.2)

**Comment by RobbinHabermehl** [comment]
Somehow I didn't receive the comments on this issue; but thanks for researching this @KevinBodi and @bjoerge, thanks for fixing it! I'll give this a try next week.
